demo/start.sh

```diff
@@ -56,26 +62,32 @@ for port in 8080 8081 8082; do
 		        compress: false
 		PORTLISTENERS
 		)
-        echo "${listeners}" >> $DIR/etc/$port.config
+        echo "${listeners}" >> "$DIR"/etc/$port.config
 
         # Disable tls for the servers
-        printf '\n\n# Disable tls on the servers.' >> $DIR/etc/$port.config
-        echo '# DO NOT USE IN PRODUCTION' >> $DIR/etc/$port.config
-        echo 'use_insecure_ssl_client_just_for_testing_do_not_use: true' >> $DIR/etc/$port.config
-        echo 'federation_verify_certificates: false' >> $DIR/etc/$port.config
+        {
+          printf '\n\n# Disable tls on the servers.'
+          echo '# DO NOT USE IN PRODUCTION'
+          echo 'use_insecure_ssl_client_just_for_testing_do_not_use: true'
+          echo 'federation_verify_certificates: false'
+        } >> "$DIR"/etc/$port.config
 
         # Set tls paths
-        echo "tls_certificate_path: \"$DIR/etc/localhost:$https_port.tls.crt\"" >> $DIR/etc/$port.config
-        echo "tls_private_key_path: \"$DIR/etc/localhost:$https_port.tls.key\"" >> $DIR/etc/$port.config
+        {
+          echo "tls_certificate_path: \"$DIR/etc/localhost:$https_port.tls.crt\""
+          echo "tls_private_key_path: \"$DIR/etc/localhost:$https_port.tls.key\""
+        } >> "$DIR"/etc/$port.config
 
         # Generate tls keys
-        openssl req -x509 -newkey rsa:4096 -keyout $DIR/etc/localhost\:$https_port.tls.key -out $DIR/etc/localhost\:$https_port.tls.crt -days 365 -nodes -subj "/O=matrix"
+        openssl req -x509 -newkey rsa:4096 -keyout "$DIR"/etc/localhost"$(printf '\:')"$https_port.tls.key -out "$DIR"/etc/localhost"$(printf '\:')"$https_port.tls.crt -days 365 -nodes -subj "/O=matrix"
```

Comment:
I assume you've tested this? It doesn't look like it will do the right thing to me.